src/Credentials/CredentialProvider.php

```diff
@@ -63,8 +68,9 @@ public static function defaultProvider(array $config = [])
     {
         return self::memoize(
             self::chain(
-                self::env(),
-                self::ini(),
+                self::cache($config),
```

Comment:
Why cache the credentials by default? This only seems valuable if you know that you are using instance profile credentials.